autopush/websocket.py

```diff
@@ -628,3 +650,9 @@ def put(self, uaid):
         client.process_notifications()
         settings.metrics.increment("updates.notification.checking")
         self.write("Notification check started")
+
+    def delete(self, uaid):
+        client = self.ap_settings.clients.get(uaid)
+        if client:
+            client.onClose(False, 0, "duplication")
```

Comment:
This calls the clients onClose handler, but the onClose func is just something the protocol calls in the course of shutting down the connection, calling it won't actually shutdown the connection. It should call client.sendClose() to start the shutdown of the connection, during the actual close sendClose will eventually call the onClose function.